app/controllers/concerns/api/v2/parameter_deprecation.rb

```diff
@@ -0,0 +1,61 @@
+# This module adds a pre-action filter that will handle deprecated properties.
```

Comment:
Sounds to me like a good reason to go with a 5 line handler that may have something in common with other cases, instead of trying to force a complex generalization.